app/lib/features/room/widgets/room_chip.dart

```diff
@@ -25,64 +24,65 @@ class RoomChip extends ConsumerStatefulWidget {
 }
 
 class _RoomChipState extends ConsumerState<RoomChip> {
-  late Uri? uri;
+  late RoomPreviewQuery query;
 
   @override
   void initState() {
     super.initState();
-    uri = widget.uri.map((u) => Uri.tryParse(u));
+    query = updateQuery();
   }
 
-  RoomPreviewQuery get query => (
-    roomIdOrAlias: widget.roomId,
-    serverNames: AllHashed(uri?.queryParametersAll['via'] ?? []),
-  );
+  @override
+  void didUpdateWidget(RoomChip oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.uri != widget.uri || widget.roomId != oldWidget.roomId) {
+      setState(() {
+        query = updateQuery();
+      }); // refresh after
+    }
+  }
+
+  RoomPreviewQuery updateQuery() {
+    return (
+      roomIdOrAlias: widget.roomId,
+      serverNames: AllHashed(widget.uri?.queryParametersAll['via'] ?? []),
+    );
+  }
```

Comment:
queries are a bit complicated, don't do them _all the time_ for every render, but keep the proper state around.